wp-content/civi-extensions/civirazorpay/api/v3/Civirazorpay/CivicrmRazorpayFetchSettlementDateCron.php

```
@@ -0,0 +1,120 @@
+
+    \Civi::log()->info("Processed " . count($returnValues) . " transactions for settlement: {$targetSettlementId}");
+    return civicrm_api3_create_success($returnValues, $params, 'Civirazorpay', 'CivicrmRazorpayFetchSettlementDateCron');
+
+  } catch (Exception $e) {
+    \Civi::log()->error("Error in Razorpay settlement cron: " . $e->getMessage(), [
+      'target_date' => $targetDate ?? 'not set',
+      'stack_trace' => $e->getTraceAsString(),
+    ]);
+    throw new CiviCRM_API3_Exception("Failed to process settlements: " . $e->getMessage(), 1001);
+  }
+}
```

PHPMD (2.15.0)

The function civicrm_api3_civirazorpay_civicrmrazorpayfetchsettlementdatecron() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10. (Code Size Rules)

(CyclomaticComplexity)